src/qmflows/packages/package_wrapper.py

```diff
@@ -119,6 +119,8 @@
 
 __all__ = ['PackageWrapper']
 
+JT = TypeVar("JT", bound=plams.core.basejob.Job)
```

Comment:
Though I generally agree with types, I am concerned with the use of complex types in Python. It is not only in this library, it is that Python typing is becoming more and more complex (like C++ templates)

Reply:
Something to consider might be to move the annotations to `.pyi` stub files,
this should atleast help to keep the main code a bit cleaner in the case of the more complicated annotations.

Reply:
I don't disagree about your type annotations, I think that they are super useful. It is just that I have the feeling that for newcomers to the language/library this may difficult to swallow. It is a philosophical comment about types in a dynamic language like Python.

Reply:
Yeah, I don't disagree with this sentiment.
I mean, it is possible for a number of functions to provide a set of compact, incredibly obvious annotations,
but more often than not the dynamic nature makes this rather challenging.

This was also my reasoning behind https://github.com/SCM-NV/qmflows/pull/226#discussion_r532733645; as this could (arguably?) result in a clearer distinction between the realm of the scary type checkers and that of the humans. Just something to consider.